pysatSpaceWeather/instruments/sw_flare.py

```diff
@@ -0,0 +1,375 @@
+    - 'prediction' Predictions from SWPC for the next 3 days
+
+Examples
+--------
+Download and load the most recent flare predictions. To always ensure you are
+loading the most recent data, load the data with tomorrow's date.
+::
+
+    flare = pysat.Instrument('sw', 'flare', tag='prediction')
+    flare.download(start=flare.tomorrow())
+    flare.load(date=flare.tomorrow())
+
+
+Note
+----
+The forecast data is stored by generation date, where each file contains the
+forecast for the next three days. Forecast data downloads are only supported
+for the current day. When loading forecast data, the date specified with the
+load command is the date the forecast was generated. The data loaded will span
+three days. To always ensure you are loading the most recent data, load
+the data with tomorrow's date.
+
+
+Warnings
+--------
+The 'prediction' flare data loads three days at a time. Loading multiple files,
+loading multiple days, the data padding feature, and multi_file_day feature
+available from the pyast.Instrument object is not appropriate for 'prediction'
+data.
+
+"""
+
+import datetime as dt
+import numpy as np
+import os
+import pandas as pds
+import pysat
+
+from pysatSpaceWeather.instruments import methods
+
+logger = pysat.logger
+
+# ----------------------------------------------------------------------------
+# Instrument attributes
+
+platform = 'sw'
+name = 'flare'
+tags = {'prelim': 'Preliminary SWPC daily solar indices',
+        'daily': 'Daily SWPC solar indices (contains last 30 days)',
+        'prediction': 'Predictions from SWPC for the next 3 days'}
+
+# Dict keyed by inst_id that lists supported tags for each inst_id
+inst_ids = {'': [tag for tag in tags.keys()]}
+
+# Dict keyed by inst_id that lists supported tags and a good day of test data
+# generate todays date to support loading forecast data
+now = dt.datetime.utcnow()
+today = dt.datetime(now.year, now.month, now.day)
+tomorrow = today + dt.timedelta(days=1)
+
+# ----------------------------------------------------------------------------
+# Instrument test attributes
+
+_test_dates = {'': {'prelim': dt.datetime(2009, 1, 1),
+                    'daily': tomorrow,
+                    'prediction': tomorrow}}
+
+# Other tags assumed to be True
+_test_download_ci = {'': {'prelim': False}}
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+preprocess = methods.general.preprocess
+
+
+def init(self):
+    """Initialize the Instrument object with instrument specific values."""
+
+    # Set the required Instrument attributes
+    self.acknowledgements = methods.swpc.ackn
+    self.references = "".join(["Check SWPC for appropriate references: ",
+                               "https://www.swpc.noaa.gov/phenomena"])
+    logger.info(self.acknowledgements)
+
+    return
+
+
+def clean(self):
+    """Clean the F10.7 data, empty function as this is not necessary."""
+
```

Comment:
For instruments we've used the logger when there is no cleaning. I can get this may not be needed for an index but the lack of cleaning should be noted somewhere. Perhaps the docstring. The same comment applies to the other indices.